pydicom/values.py

```diff
@@ -379,20 +379,18 @@ def convert_value(VR, raw_data_element, encoding=default_encoding):
         logger.debug('unable to translate tag %s with VR %s'
                      % (raw_data_element.tag, VR))
 
-        for convert_vr in convert_retry_VR_order:
-            vr = convert_vr
-            converter = converters[vr]
+        for vr in convert_retry_VR_order:
             if vr == VR:
                 continue
             try:
                 value = convert_value(vr, raw_data_element, encoding)
-                break
-            except Exception:
-                pass
-            else:
                 logger.debug('converted tag %s with VR %s'
                              % (raw_data_element.tag, vr))
-                value = raw_data_element.value
+                return value
```

Comment:
The test case for this is in BadValueReadtests in test_valuerep.py

better test coverage might be with:

```
class A(object):
    pass
a = A()
a.value = "1A"
a.is_little_endian = True
a.is_implicit_VR = False
a.tag = (0x0010, 0x0020)
from pydicom.values import *
import pydicom.config

convert_value('SH', a) == "1A"  # all good
convert_value('IS', a) ==  "1A"  # converted with fallback vr "SH"
pydicom.values.convert_retry_VR_order = ['FL', 'UL']
convert_value('IS', a) == "1A"  # no fallback VR succeeded, returned original value untranslated.

pydicom.config.enforce_valid_values = True
convert_value('SH', a) == "1A"  # all good
raises(convert_value('IS', a), ValueError)  # invalid literal for base 10
```